src/solvers/petsc_nonlinear_solver.C

```diff
@@ -128,12 +134,21 @@ extern "C"
     return 0;
   }
 
+#ifdef LIBMESH_ENABLE_DEPRECATED
+  PetscErrorCode
+  __libmesh_petsc_snes_monitor (SNES, PetscInt its, PetscReal fnorm, void *)
+  {
+    libmesh_deprecated();
+    libmesh_petsc_snes_monitor(nullptr, its, fnorm, nullptr);
```

Comment:
This callback returns an actual error code, so why are we always returning 0?

Reply:
Oh, I see now that the error code returned by the original version of `__libmesh_petsc_snes_monitor` was always 0. This version should simply 
```
return libmesh_petsc_snes_monitor(nullptr, its, fnorm, nullptr);
```
though.